tests/test_ruby_server.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# Any copyright is dedicated to the Public Domain.
+# http://creativecommons.org/publicdomain/zero/1.0/
+
+from pathlib import Path
+
+import glean_parser
+from glean_parser import translate
+
+ROOT = Path(__file__).parent
+
+
+def test_parser_rb_server_ping_file(tmpdir, capsys):
+    """Test that no files are generated if only ping definition
+    is provided."""
+    tmpdir = Path(str(tmpdir))
+
+    translate.translate(
+        [
+            ROOT / "data" / "ruby_server_metrics.yaml",
+            ROOT / "data" / "ruby_server_pings.yaml",
+        ],
+        "ruby_server",
+        tmpdir,
+    )
+    captured = capsys.readouterr()
+    assert all(False for _ in tmpdir.iterdir())
+    assert (
+        "ping definition found. Server-side environment is simplified" in captured.out
+    )
+
+
+def test_parser_rb_server_no_event_metrics(tmpdir, capsys):
+    """Test that no files are generated if no event metrics."""
+    tmpdir = Path(str(tmpdir))
+
+    translate.translate(
+        [ROOT / "data" / "ruby_server_no_events.yaml"],
+        "ruby_server",
+        tmpdir,
+    )
+    captured = capsys.readouterr()
+    assert all(False for _ in tmpdir.iterdir())
+    assert "no events found...at least one event metric is required" in captured.out
+
+
+def test_parser_rb_server_metrics_unsupported_type(tmpdir, capsys):
+    """Test that no files are generated if only metric definitions
+    are provided without pings."""
```

Comment:
This doc string seems incorrect, given the test name and test file.

Reply:
fixed all test docstrings